common/proof_test.go

```diff
@@ -2,14 +2,163 @@ package common_test
 
```

Comment:
Same comment for this file.
I think it would be proper to use `require` package to handle error instead of `log.Fatal`